openedx/core/djangoapps/user_api/accounts/tests/test_views.py

```diff
@@ -562,3 +565,48 @@ def test_convert_relative_profile_url(self):
                 "image_url_small": "http://testserver/profile_images/default_10.jpg"
             }
         )
+
+    @ddt.data(
+        ("client", "user", True),
+        ("different_client", "different_user", False),
+        ("staff_client", "staff_user", True),
+    )
+    @ddt.unpack
+    def test_parental_consent(self, api_client, requesting_username, has_full_access):
+        """
+        Verifies that under thirteens never return a public profile.
+        """
+        client = self.login_client(api_client, requesting_username)
+
+        # Set the user to be ten years old with a public profile
+        legacy_profile = UserProfile.objects.get(id=self.user.id)
+        current_year = datetime.datetime.now().year
+        legacy_profile.year_of_birth = current_year - 10
+        legacy_profile.save()
+        set_user_preference(self.user, ACCOUNT_VISIBILITY_PREF_KEY, ALL_USERS_VISIBILITY)
+
+        # Verify that the default view is still private (except for clients with full access)
+        response = self.send_get(client)
+        if has_full_access:
+            data = response.data
```

Comment:
There is, but that assumes all the default settings. I considered refactoring out the similar helper method in one of the other tests, but even then the year of birth and requires parental consent values are different.